packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -396,7 +400,7 @@ export class DeltaManager
             });
 
         this._inbound.on("error", (error) => {
-            this.close(CreateContainerError(error));
+            this.close(CreateCorruptionError(error, {}));
```

Comment:
It would be great to have some write-up on what you are trying to do and why you are choosing this particular path over some other.

Some things that look to be potentially wrong:
1. This will capture exceptions on op processing. Those exceptions may not be data corruption issues. I.e. it can be an exception coming from some event listener (that we do not control) and it's possible it will not happen again when user refreshes document (starts over). So it's a crash, but not permanent data loss.
2. There are other paths we get to closing container. Why this code is here, as opposed to being in Container.close() method?
3. The logic you have in CreateCorruptionError is partially duplication of logic of what we have in CreateContainerError & TelemetryLogger.prepareErrorObject. The most places we do these things, the more likely we will make mistake somewhere (of fail to fix a bug when it occurred).

Clarifying these points in overall design description would be great, as it will help better understand if things are in right places or not

Reply:
Thanks @vladsud , I've updated the top post.  In regards to your questions

1. We were comfortable* that false positives would be preferable to false negatives -- at least for the time being -- I believe we are both open to "tweaking" the behavior here (@anthony-murphy if we revert deltaManager.ts entirely, what categories of errors might we miss by guarding just the op processors?)
2. I think the decision may be somewhat arbitrary.  Perhaps `Container.close()` is a better place (if the intent expressed here is worth preserving)?
3. How do you feel about some utility method that looks like `type ExtractsLoggables = (error: any) => (undefined | Partial<{ message: string, stack: string, errorType: string }>)` to be shared among the call sites?

---

(*) ...not *totally* comfortable -- to be frank -- per the discussion [above](https://github.com/microsoft/FluidFramework/pull/5144#pullrequestreview-591512734)